.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "submodules/FileSystem"]
+	path = submodules/FileSystem
+	url = git@github.com:aspnet/FileSystem.git
```

Comment:
Yeah, we should use https. You only ever need creds if you push from the submodule. This is a known issue with submodules, users can't customize the protocol they use. It sucks but hey, you can always use a credential helper like [this fantastic specimen from some amazing developer](https://gitcredentialstore.codeplex.com/) :)